react/utils.js

```diff
@@ -151,13 +151,29 @@ export function hasPostalCodeGeocoordinates(address, searchAddress) {
 }
 
 export function findSlaWithChannel(item, channel) {
-  return (
-    item &&
-    item.slas &&
-    item.slas.find(sla =>
-      hasOnlyScheduledDelivery(item.slas, channel)
-        ? isCurrentChannel(sla, channel)
-        : isCurrentChannel(sla, channel) && !hasDeliveryWindows(sla)
-    )
+  if (!item || !item.slas) {
+    return null
+  }
+
+  const hasOnlyScheduledDeliverySla = hasOnlyScheduledDelivery(
+    item.slas,
+    channel
   )
+
+  const currentChannelSlas = item.slas.filter(sla => {
+    return (
+      isCurrentChannel(sla, channel) &&
+      (!hasDeliveryWindows(sla) || hasOnlyScheduledDeliverySla)
+    )
+  })
+
+  let cheapestSla = !!currentChannelSlas.length && currentChannelSlas[0]
```

Comment:
you don't need this
```suggestion
  let cheapestSla = currentChannelSlas.length && currentChannelSlas[0]
```

Reply:
actually, this could be just
```suggestion
  let cheapestSla = currentChannelSlas[0]
```
if the array is empty, `cheapestSla` will be `undefined`

Reply:
Nicee! Done.